### PR TITLE
Correctly use toNumber() and toString() for BigNumber

### PR DESCRIPTION
**Pull request by adamdossa**
Currently test fails on a new testrpc network (so all accounts have 100ETH):
```
  1) Contract: Conference on setLimitOfParticipants returns only your deposit for multiple invalidations:
     AssertionError: expected false to equal true
      at test/conference.js:68:16
      at process._tickCallback (internal/process/next_tick.js:103:7)

```
This is because the assert relies on comparing the "raw" BigNumber objects, not the underlying numbers and ">" may behave unexpectedly on the raw object. In this case the two BigNumber objects are:
```
{ [String: '100000000000000000000'] s: 1, e: 20, c: [ 1000000 ] }
{ [String: '99996975500000000000'] s: 1, e: 19, c: [ 999969, 75500000000000 ] }

```
and the ">" operator returns false.

I also fixed up the other assert operators for similar edge cases, using toString when comparing against a string, and toNumber when comparing against a number (as in the above failing case).

All tests run to success.

### CODE REVIEW

**Comment by makoto** [comment]
@adamdossa I have been wondering why the test occasionally fails. Thank you so much!!

File: test/conference.js [hunk 1]
```diff
@@ -128,13 +128,13 @@ contract('Conference', function(accounts) {
...
         meta = _meta;
-        beforeContractBalance = web3.eth.getBalance(meta.address);
+        beforeContractBalance = web3.eth.getBalance(meta.address).toNumber();
         return meta.register.sendTransaction(twitterHandle, {value:transaction});
       }).then(function() {
         return meta.totalBalance.call();
       })
       .then(function(value){
-        assert.equal(value.toString() - beforeContractBalance, transaction);
+        assert.equal(value..toNumber() - beforeContractBalance, transaction);
```
Comment (makoto) [inline test/conference.js]: BTW, found the typoe `..`

Comment (makoto) [inline test/conference.js]: I fixed on master. Do you know if there is a way to `git co $fork_branch` without cloning the fork repo? I was too lazy to test your branch before merging, hence I had to commit the fix directly on master.
